charmcraft/config.py

```diff
@@ -328,7 +335,22 @@ def unmarshal(cls, obj: Dict[str, Any], project: Project):
             # base configurations.  Doing it here rather than a Union
             # type will simplify user facing errors.
             bases = obj.get("bases")
-            if bases is not None and isinstance(bases, list):
+            if bases is None:
+                # Set default bases to Ubuntu 20.04 to match strict snap's
+                # effective behavior.
+                bases = [
+                    {
+                        "name": "ubuntu",
+                        "channel": "20.04",
+                        "architectures": [get_host_architecture()],
+                    }
+                ]
+                if not is_charmcraft_running_in_managed_mode():
```

Comment:
Same here. Why we don't want the message in managed mode?

Reply:
this is to not have the message printed twice, once from the outer shell and later in the inner run (maybe a code comment can help)

Reply:
Exactly

Reply:
Do we want *any* deprecation message in managed mode? Maybe the correct place for this "if" is inside the `notify_deprecation` function.

Reply:
Oh, that's a better idea, I'll do that!

Reply:
Done!